src/query/service/src/api/rpc/exchange/exchange_manager.rs

```diff
@@ -821,16 +883,53 @@ impl FragmentCoordinator {
             self.initialized = true;
 
             let pipeline_ctx = QueryContext::create_from(ctx);
+
             let pipeline_builder = PipelineBuilder::create(
                 pipeline_ctx.get_function_context()?,
                 pipeline_ctx.get_settings(),
                 pipeline_ctx,
                 enable_profiling,
                 SharedProcessorProfiles::default(),
             );
-            self.pipeline_build_res = Some(pipeline_builder.finalize(&self.physical_plan)?);
+            let res = pipeline_builder.finalize(&self.physical_plan)?;
+
+            self.pipeline_build_res = Some(res);
         }
 
         Ok(())
     }
+
+    pub fn prepare_pipeline_local(
```

Comment:
Why need this?

Reply:
because we need to give HashJoinBuildState from first pipeline builder to next pipeline builder. https://github.com/datafuselabs/databend/pull/13151/files#diff-d848f447208973291e64f8bd6c7cb3e0b3dea5ff3a8459327180ecd1936631adR647